src/components/shared/TinderCard/index.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  Animated,
+  Dimensions,
+  LayoutAnimation,
+  PanResponder,
+  Platform,
+  Text,
+  UIManager,
+} from 'react-native';
+import React, { ReactElement, useCallback, useMemo, useState } from 'react';
+
+import styled from 'styled-components/native';
+
+const SCREEN_WIDTH = Dimensions.get('screen').width;
+const SWIPE_THRESHOLD = 0.25 * SCREEN_WIDTH;
+const SWIPE_OUT_DURATION = 250;
+
+interface DataProps {
+  [key: string]: any;
+}
+
+interface Props {
+  testID?: string;
+  onSwipeRight?: (item: DataProps) => void;
+  onSwipeLeft?: (item: DataProps) => void;
+  onCancel?: () => void;
+  data: Array<DataProps>;
+  renderCards: (item: DataProps) => ReactElement;
+  renderNoMoreCards: () => ReactElement;
+  rotate?: boolean;
+  stackSize?: number;
+}
+
+const Container = styled.View`
+  width: ${SCREEN_WIDTH};
+`;
+
+const CancelButton = styled.Button`
+  background-color: blue;
+  color: blue;
+`;
+
+const NoCard = styled.View`
+  width: 100%;
+  height: 200px;
+  justify-content: center;
+  align-items: center;
+  padding: 10px;
+`;
+
+function TinderCard(props: Props): ReactElement {
+  const [cardIndex, setCardIndex] = useState(0);
+  const position = useMemo(() => new Animated.ValueXY(), []);
+
+  const resetPosition = (): void => {
+    Animated.spring(position, {
+      toValue: { x: 0, y: 0 },
+    }).start();
+  };
+
+  const onSwipeCompleted = useCallback((direction: string): void => {
+    const { onSwipeLeft, onSwipeRight, data } = props;
+
+    position.setValue({ x: 0, y: 0 });
+
+    UIManager.setLayoutAnimationEnabledExperimental &&
+      UIManager.setLayoutAnimationEnabledExperimental(true);
+    LayoutAnimation.spring();
+
+    let currentIndex = 0;
+
+    setCardIndex((idx) => {
+      currentIndex = idx;
+      return idx + 1;
+    });
+
+    if (direction === 'right') {
```

Comment:
Using `string` with if statement causes various side effects. Please try to use things like boolean or enum in this case.